client/mds3/test/mds3.integration.spec.js

```diff
@@ -400,6 +405,81 @@ export async function testVariantLeftLabel(test, tk, bb) {
 	}
 }
 
+tape.only('Official - allow2selectSamples', test => {
+	testAllow2selectSamples('hg38-test', 'tp53', 'TermdbTest', test)
+})
+
+export async function testAllow2selectSamples(genome, gene, dslabel, test) {
+	/* 
+must use a gene with both single and multi occurrence mutations to test
+*/
+	const holder = getHolder()
+	const buttonText = 'SElect sample'
+	runproteinpaint({
+		holder,
+		genome,
+		gene,
+		tracks: [
+			{
+				type: 'mds3',
+				dslabel,
+				callbackOnRender,
+				allow2selectSamples: {
+					buttonText, // hardcoded text should show in selection button
+					callback: () => {} // TODO figure out a way to verify callback is called by clicking button
+				}
+			}
+		]
+	})
+	async function callbackOnRender(tk, bb) {
+		// 1: click on singleton mutation to show selection button in menu
+		const singletonMutationDisc = tk.skewer.g
+			.selectAll('.sja_aa_disckick')
+			.nodes()
+			.find(i => i.__data__.occurrence == 1)
+		singletonMutationDisc.dispatchEvent(new Event('click'))
+		await whenVisible(tk.itemtip.d)
+		{
+			//const buttons = tk.itemtip.d.selectAll('button').nodes()
+			const buttons = await detectGte({ elem: tk.itemtip.d.node(), selector: 'button', count: 1 })
+			test.equal(buttons[0].innerHTML, buttonText, '1st button in single-sample menu shows buttonText')
+			// TODO trigger click on selection button
+		}
+
+		// 2: click on multi-sample mutation to show selection button in sample table
+		const multiMutationDisc = tk.skewer.g
+			.selectAll('.sja_aa_disckick')
+			.nodes()
+			.find(i => i.__data__.occurrence > 1)
+		multiMutationDisc.dispatchEvent(new Event('click'))
+		await whenVisible(tk.itemtip.d)
+		{
+			const buttons = tk.itemtip.d.selectAll('button').nodes()
+			test.equal(buttons[buttons.length - 1].innerHTML, buttonText, 'last button in multi-sample menu shows buttonText')
+			test.ok(buttons[buttons.length - 1].disabled, 'button is also disabled (when no checkbox is checked)')
+			// must check one checkbox first to
+		}
+
+		// 3: click on sample leftlabel to show selection button in sample table
+		tk.leftlabels.doms.samples.node().dispatchEvent(new Event('click'))
+		await whenVisible(tk.menutip.d)
+		{
+			const btn = await detectOne({ elem: tk.menutip.dnode, selector: '.sja_mds3_slb_sampletablebtn' })
+			btn.dispatchEvent(new Event('click'))
+			await detectOne({ elem: tk.menutip.dnode, selector: 'table' }) // wait when table is shown
+			const buttons = tk.menutip.d.selectAll('button').nodes()
+			test.equal(buttons[buttons.length - 1].innerHTML, buttonText, 'last button in sample table shows buttonText')
+			test.ok(buttons[buttons.length - 1].disabled, 'button is also disabled (when no checkbox is checked)')
+		}
+		if (test._ok) {
+			tk.menutip.d.remove()
+			tk.itemtip.d.remove()
+			holder.remove()
```

Comment:
@xzhou82 It would be good later to have a component/app/tk `.destroy()` method to handle this cleanup that could be called here or in a framework or callback. In the destroy method it's also good to delete references to these deleted selections/objects, like `for(const key in app.dom) delete app.dom[key]` or similar.